src/node/node_state.h

```diff
@@ -1317,96 +1317,118 @@ namespace ccf
       // inform the host of any nodes that no longer need to be tracked.
       network.nodes.set_global_hook(
         [this](kv::Version version, const Nodes::Write& w) {
-          for (auto& [node_id, ni] : w)
+          for (const auto& [node_id, opt_ni] : w)
           {
-            if (ni.value.status == NodeStatus::RETIRED)
+            if (!opt_ni.has_value())
+            {
+              throw std::logic_error("Unexpected: removal from nodes table");
```

Comment:
Not likely this'll happen, but if it does there's a small chance we may be happy to have the node id.